centreon/www/class/centreonDB.class.php

```diff
@@ -44,110 +44,79 @@
 require_once __DIR__ . '/centreonLog.class.php';
 
 /**
- * Class CentreonDB used to manage DB connection
+ * Class
+ *
+ * @class CentreonDB
+ * @description used to manage DB connection
  */
-class CentreonDB extends \PDO
+class CentreonDB extends PDO
 {
-    /**
-     * @var string
-     */
+    public const DRIVER_PDO_MYSQL = "mysql";
     public const LABEL_DB_CONFIGURATION = 'centreon';
-
-    /**
-     * @var string
-     */
     public const LABEL_DB_REALTIME = 'centstorage';
 
-    /**
-     * @var array<string,\CentreonDB>
-     */
-    private static $instance = [];
-
-    /**
-     * @var string
-     */
-    protected $db_type = "mysql";
+    /** @var int */
+    private const RETRY = 3;
 
-    /**
-     * @var string
-     */
-    protected $db_port = "3306";
+    /** @var array<string,CentreonDB> */
+    private static $instance = [];
 
-    /**
-     * @var int
-     */
+    /** @var int */
     protected $retry;
 
-    /**
-     * @var array<string,mixed>
-     */
-    protected $dsn;
-
-    /**
-     * @var array<int, array<int, mixed>|int|bool|string>
-     */
+    /** @var array<int, array<int, mixed>|int|bool|string> */
     protected $options;
 
-    /**
-     * @var string
-     */
+    /** @var string */
     protected $centreon_path;
 
-    /**
-     * @var \CentreonLog
-     */
-    protected $log;
-
+    /** @var CentreonLog */
+    protected $logger;
 
     /*
      * Statistics
      */
 
-    /**
-     * @var int
-     */
+    /** @var int */
     protected $requestExecuted;
 
-    /**
-     * @var int
-     */
+    /** @var int */
     protected $requestSuccessful;
 
-    /**
-     * @var int
-     */
+    /** @var int */
     protected $lineRead;
 
-    /**
-     * @var int
-     */
+    /** @var int */
     private $queryNumber;
 
-    /**
-     * @var int
-     */
+    /** @var int */
     private $successQueryNumber;
 
     /**
      * Constructor
      *
-     * @param string $db | centreon, centstorage
+     * @param string $dbLabel LABEL_DB_* constants
      * @param int $retry
-     *
+     * @param CentreonDbConfig|null $dbConfig
      * @throws Exception
      */
-    public function __construct($db = self::LABEL_DB_CONFIGURATION, $retry = 3)
-    {
+    public function __construct(
+        $dbLabel = self::LABEL_DB_CONFIGURATION,
+        $retry = self::RETRY,
+        private ?CentreonDbConfig $dbConfig = null
```

Comment:
```suggestion
        private CentreonDbConfig $dbConfig = null
```

You can provide a null value, but after construction, the property cannot be.
Another question is: do we need to keep the config object as property of this class or can it be an object used to ease usage but only inner values are stored as properties?
Only `$this->dbConfig->dbName` is used outside the constructor